plugins/woocommerce/includes/admin/class-wc-admin-notices.php

```diff
@@ -115,6 +119,46 @@ public static function reset_admin_notices() {
 		self::add_notice( 'template_files' );
 		self::add_min_version_notice();
 		self::add_maxmind_missing_license_key_notice();
+		self::maybe_add_legacy_api_removal_notice();
+	}
+
+	/**
+	 * Add an admin notice about the removal of the Legacy REST API if the said API is enabled.
+	 *
+	 * TODO: Remove this method in WooCommerce 9.0.
```

Comment:
```suggestion
	 * TODO: Remove this method in WooCommerce 9.2.
```

We might want to keep this notice (or rather a modified version of it) around for a couple of versions after 9.0, in case a site somehow misses the notice beforehand.

Reply:
Yes and no. I think what we should do in 9.0 is to replace the admin notice with another saying "The legacy REST API has been removed" and pointing to the new extension, but only on sites where it was enabled and the plugin hasn't been installed (and we are unable to auto-install it, if we decide to try that).

Reply:
👍